packages/stylelint-plugin/src/codemod.spec.ts

```diff
@@ -0,0 +1,523 @@
+    expectedUnfixables: 0,
+  },
+  {
+    language: "scss",
+    testName:
+      "negation of old token is detected as an equation, and then fixed with a calc(), but not if it's already within one",
+    input:
+      '@import "~@kaizen/design-tokens/sass/spacing"; .foo { padding: calc(-$kz-spacing-lg); }',
+    expectedOutput:
+      '@import "~@kaizen/design-tokens/sass/spacing-vars"; .foo { padding: calc((-1 * #{$kz-var-spacing-lg})); }',
+    expectedUnfixables: 0,
+  },
+  {
+    language: "scss",
+    testName:
+      "negation of old token is detected as an equation, and then fixed with a calc(), even when it's part of a value with multiple 'sides' ",
+    input:
+      '@import "~@kaizen/design-tokens/sass/spacing"; .foo { padding: 5px -$kz-spacing-lg; }',
+    expectedOutput:
+      '@import "~@kaizen/design-tokens/sass/spacing-vars"; .foo { padding: 5px calc(-1 * #{$kz-var-spacing-lg}); }',
+    expectedUnfixables: 0,
+  },
+  {
+    language: "scss",
+    testName:
+      "negation of old token is detected as an equation, and then fixed with a calc(), even when it's part of a value with multiple 'sides', and with another kaizen token next to it ",
+    input:
+      '@import "~@kaizen/design-tokens/sass/spacing"; .foo { padding: $kz-spacing-md -$kz-spacing-lg; }',
+    expectedOutput:
+      '@import "~@kaizen/design-tokens/sass/spacing-vars"; .foo { padding: $kz-var-spacing-md calc(-1 * #{$kz-var-spacing-lg}); }',
+    expectedUnfixables: 0,
+  },
+  {
+    language: "scss",
+    testName:
+      "doesn't fix ambiguous case of negation (knows if it's an equation and not a negation - all it takes is a space)",
+    input:
+      '@import "~@kaizen/design-tokens/sass/spacing"; .foo { padding: 5px - $kz-spacing-lg; }',
+    expectedOutput:
+      '@import "~@kaizen/design-tokens/sass/spacing"; .foo { padding: 5px - $kz-spacing-lg; }',
+    expectedUnfixables: 2,
+  },
+
+  {
+    language: "less",
+    testName:
+      "negation of old token is detected as an equation, but not fixed in LESS",
+    input:
+      '@import "~@kaizen/design-tokens/less/spacing"; .foo { padding: -@kz-spacing-lg; }',
+    expectedOutput:
+      '@import "~@kaizen/design-tokens/less/spacing"; .foo { padding: -@kz-spacing-lg; }',
+    expectedUnfixables: 1,
+  },
+  {
+    language: "scss",
+    testName:
+      "fixes variables even though next to an operator but separated by a comma",
+    input:
+      '@import "~@kaizen/design-tokens/sass/spacing"; .foo { padding: rgba(+, $kz-spacing-md) }',
+    expectedOutput:
+      '@import "~@kaizen/design-tokens/sass/spacing-vars"; .foo { padding: rgba(+, $kz-var-spacing-md) }',
+    expectedUnfixables: 0,
+  },
+  {
+    language: "scss",
+    testName: "compiles mix() function in place",
+    input:
+      '@import "~@kaizen/design-tokens/sass/color"; $customVar: white; .foo { color: mix($kz-color-wisteria-800, $customVar, 80%) }',
+    expectedOutput: "$customVar: white; .foo { color: #5d5f6e }",
```

Comment:
Hey this is a really good point.

Because we already ran this on murmur and perf-ui it is unfortunately a little too late.
We do have git history/annotations though that i assumed would be of help when going back to fix them.

The other thing is, when going through a bunch of these deprecated colors in Murmur with Marc, the only thing that he was really interested in is the context around the color and the semantics (e.g. "this looks like the background of a primary button, it should be X, and it's so old anyway so let's not stress too much about precision").

Reply:
Following up in this PR https://github.com/cultureamp/kaizen-design-system/pull/1624